documentation/disaster-recovery-testing.md

```diff
@@ -0,0 +1,208 @@
+
+## Documentation requirements
+
+Copy the template DR testing document which will be a record of the steps run, time taken, and any issues
+- template doc tba
+
+## Initial set-up
+
+Participants must have access to Github and the repositories.
+
+Schedule virtual meeting for the test to take place
+- teams or slack
+- invite relevant technical stakeholders
+
+Regularly provide updates on the service Slack channel to keep product owners abreast of developments.
+
+## Freeze pipeline
+
+Alert developers that no one should merge to main.
+- In github setings, a user with repo admin privileges should update the Branch protection rules and set required PR approvers to 6
+
+
+## Follow instructions for the selected scenario
+
+
+## Scenario 1. Loss of database instance
+
+In case the database instance is lost, the objectives are:
+
+- Recreate the lost postgres database instance
+- Restore data from nightly backup stored in Azure.
+
+The point-in-time and snapshot backups created by the Azure Postgres service may not be available if it's been deleted, or if there is an Azure region issue.
+
+### Delete the postgres database instance
+
+Note that you must have a previously created backup on azure storage before starting this step. If not, create one now before continuing.
+
+- Delete the existing postgres database
+    - manually delete via UI https://portal.azure.com/#browse/Microsoft.DBforPostgreSQL%2FflexibleServers
+- confirm it's deleted
+
+### Enable maintenance mode
+
+Run the enable maintenance workflow for the service and environment affected
+
+### Recreate the lost postgres database instance
+
+Run the deploy workflow to recreate the missing postgres database (and any other missing resources).
+
+As the maintenance page has been implemented, you will need to
+- create a branch from main
+- update the env tfvars.json as per https://github.com/DFE-Digital/teacher-services-cloud/blob/main/documentation/maintenance-page.md#configure-terraform-to-keep-deploying-the-application
+- push the branch to github
+- run the deploy workflow using your branch for the env being tested
+
+The web app may fail to start due to the postgres db missing data. It needs to be running so that the restore workflow can connect to the database using konduit. If it is failing, update your branch
+- override the start command to remove any db commands or a dummy command (e.g. sleep 999999).
+- or remove startup/liveness probes
+- or any other method that enables the web app to start
+- push the branch and rerun deploy workflow as above
+
+### Restore the data from previous backup in Azure storage
+
+Run the restore from backup workflow
+
+### Redeploy app if required
+
+If the start command was updated previously
+- put back the original start command in your PR
+- push to github and rerun deploy
+
+### Validate app
+
+Confirm the app is working and is using the restored data
+
```

Comment:
Requires the Disable maintenance step

Reply:
moved the Disable maintenance step before validate